### PR TITLE
[DO NOT MERGE] NS - User Mgmt - Staff Landing Pages

### PR DESCRIPTION
**Pull request by shafi-sadaindia**
### Description

- NS - User Mgmt - Staff Landing Pages

#### Screenshots
Screenshot 2023-02-24 at 2 14 47 PM
Screenshot 2023-02-24 at 2 13 40 PM

### Other Details

- [JIRA Ticket](https://collectiveshift.atlassian.net/browse/SNHUT-3503)

#### Checklist

- [x] **Assumptions of User Story met (including acceptance criteria)**
- [x] **All new necessary Unit tests were CREATED** to promote code coverage
- [x] **All required unit tests are PASSING**
- [x] The proposed functionality was tested manually locally, and it's working fine.
- [ ] PR was properly reviewed by a team member (at least 1 approval required to merge!)
- [x] Unnecessary console.logs AND/OR comments were removed
- [ ] Feature ok-ed by UX designer (If needed - not for back end components) and Product Owner
- [x] Ensure configuration and/or build changes documented

### CODE REVIEW

File: apps/toboggan-app/src/app/landing/components/landing/landing.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      providers: [ActivityMonitorService,
+      { provide: AuthService, useValue: mockAuthService },]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(LandingComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
```
Comment (sada-ajith) [inline apps/toboggan-app/src/app/landing/components/landing/landing.component.spec.ts:25]: Can you add few more test cases ?

File: apps/toboggan-app/src/app/landing/components/landing/landing.component.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  templateUrl: './landing.component.html',
+  styleUrls: ['./landing.component.scss'],
+})
+export class LandingComponent implements OnInit  {
+  landingLinks : any = [] ;
+  userType : string | null | undefined;
+  constructor(private userService:UserService,private authService:AuthService) {}
+
+  ngOnInit(): void {
+    const currentUser = this.authService.getUserId();
```
Comment (sada-ajith) [inline apps/toboggan-app/src/app/landing/components/landing/landing.component.ts:17]: can we move this to auth service so , that if we need it in usertype somwhere else , we can reuse it ?  ( may be store it in localstorage )

File: apps/toboggan-app/src/app/landing/pages/landing-main-page/landing-main-page.component.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'toboggan-ws-landing-main-page',
+  templateUrl: './landing-main-page.component.html',
+  styleUrls: ['./landing-main-page.component.scss'],
+})
+export class LandingMainPageComponent  {
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
```
Comment (sada-ajith) [inline apps/toboggan-app/src/app/landing/pages/landing-main-page/landing-main-page.component.ts:9]: we can remove this constructor

**Review by sada-ajith — CHANGES_REQUESTED** [review]

File: apps/toboggan-app/src/app/landing/components/landing/landing.component.ts (hunk 2 above)
Comment (shafi-sadaindia) [reply apps/toboggan-app/src/app/landing/components/landing/landing.component.ts:17]: I have added getUserId method in authservice. it is calling from authservice itself.

File: apps/toboggan-app/src/app/landing/components/landing/landing.component.spec.ts (hunk 1 above)
Comment (shafi-sadaindia) [reply apps/toboggan-app/src/app/landing/components/landing/landing.component.spec.ts:34]: Done

File: apps/toboggan-app/src/app/landing/pages/landing-main-page/landing-main-page.component.ts (hunk 3 above)
Comment (shafi-sadaindia) [reply apps/toboggan-app/src/app/landing/pages/landing-main-page/landing-main-page.component.ts:9]: Done

File: apps/toboggan-app/src/app/shared/auth/auth.service.ts [hunk 4]
```diff
@@ -300,6 +302,12 @@ export class AuthService {
             status: 'success',
             message: '',
           });
+
+          const currentUser = JSON.parse(localStorage.getItem('currentUser') as any);
+          const userDetails: any = await firstValueFrom(this.userService.getUser(userSession.userId));
```
Comment (sada-ajith) [inline apps/toboggan-app/src/app/shared/auth/auth.service.ts:307]: this should part of sign in credentials as well

File: apps/toboggan-app/src/app/shared/auth/auth.service.ts [hunk 5]
```diff
@@ -332,4 +340,12 @@ export class AuthService {
       throw new Error('User session not available');
     }
   }
+
+  getUserDetails() {
```
Comment (sada-ajith) [inline apps/toboggan-app/src/app/shared/auth/auth.service.ts:344]: we already have a function for current user
Please refer
https://github.com/snhuproduct/turbine/blob/d02ddea65652c6fcfc9a40e152866a69398de3a8/apps/toboggan-app/src/app/shared/auth/auth.service.ts#L238

File: apps/toboggan-app/src/app/shared/services/user/user.service.ts [hunk 6]
```diff
@@ -19,6 +19,10 @@ export class UserService {
 
   constructor(private http: HttpClient) {}
 
+  getUser(id : string) {
```
Comment (sada-ajith) [inline apps/toboggan-app/src/app/shared/services/user/user.service.ts:22]: can we replace id with uuid. ?

**Review by sada-ajith — CHANGES_REQUESTED** [review]

**Review by sada-ajith — APPROVED** [review]

**Pull request merged**